tests/alive-tv/undef/strip-add-eq.srctgt.ll

```diff
@@ -0,0 +1,26 @@
+; TEST-ARGS: -se-verbose
+define i32 @src(i32 %x, i32 %y) {
+  %v = add i32 %x, 1
+  %c = icmp eq i32 %x, %y
+  br i1 %c, label %A, label %B
+A:
+  %w = add i32 %x, 2
+  ret i32 %w
+B:
+  %w2 = add i32 %y, 2
+  ret i32 %w2
+}
+
+define i32 @tgt(i32 %x, i32 %y) {
+  %v = add i32 %x, 1
+  %c = icmp eq i32 %x, %y
+  br i1 %c, label %A, label %B
+A:
+  %w = add i32 %x, 2
+  ret i32 %w
+B:
+  %w2 = add i32 %y, 2
+  ret i32 %w2
+}
+
+; CHECK: (ite (= %x %y) (bvadd #x00000002 %x) (bvadd #x00000002 %y)) / true
```

Comment:
this is brittle, as the order of elements can change at any time. It's controlled by Z3.
Isn't there a better way of checking that the optimization is working? maybe not, but this way isn't very good.

Reply:
Maybe I can simplify this example to return `x + 2` only - the order wouldn't matter then.

Reply:
Since it could be `2 + x` too, made it simply `x`